### PR TITLE
I862 smv result module

### PR DESCRIPTION
**Pull request by laneb**
This facilitates #862. `SmvModel` will implement `SmvResultModel`.

### CODE REVIEW

File: src/main/python/smv/smvdataset.py [hunk 1]
```diff
@@ -601,12 +610,68 @@ def run(self, i):
...
+
+            Override this method to define the output of this module, given a map
+            'i' from inputSmvDataSet to resulting DataFrame. 'i' will have a
+            mapping for each SmvDataSet listed in requiresDS. E.g.
+
+            def requiresDS(self):
+                return [MyDependency]
+
+            def run(self, i):
+                return i[MyDependency].select("importantColumn")
```
Comment (ninjapapa) [inline src/main/python/smv/smvdataset.py]: The example here is confusing. How about 
```
return train_model(i[MyDependency])
```

Comment (laneb) [inline src/main/python/smv/smvdataset.py]: This whole docstring including the example is just c/p from `SmvDataSet.run`. I need to update the contents to make them more relevant.

Comment (laneb) [inline src/main/python/smv/smvdataset.py]: Yes, forgot to update the example.

Comment (laneb) [inline src/main/python/smv/smvdataset.py]: Fixed.

**Comment by laneb** [comment]
See #869 for full PR.

**Pull request closed without merging**